streaming/base/converters/dataframeToMDS.py

```diff
@@ -0,0 +1,251 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""A utility to convert databricks' tables to MDS."""
+
+import json
+import os
+import shutil
+from argparse import ArgumentParser
+from collections.abc import Iterable
+from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple, Union
+
+import pandas as pd
+from pyspark import TaskContext
+from pyspark.sql import SparkSession
+from pyspark.sql.dataframe import DataFrame
+from pyspark.sql.types import StringType, StructField, StructType
+
+from streaming import MDSWriter
+from streaming.base.storage.upload import CloudUploader
+
+default_mds_kwargs = {
+    'compression': 'zstd:7',
+    'hashes': ['sha1', 'xxh64'],
+    'size_limit': 1 << 27,
+    'progress_bar': 1,
+    'columns': {
+        'tokens': 'bytes'
+    },
+    'keep_local': False
+}
+
+default_udf_kwargs = {
+    'concat_tokens': 2048,
+    'tokenizer': 'EleutherAI/gpt-neox-20b',
+    'eos_text': '<|endoftext|>',
+    'compression': 'zstd',
+    'split': 'train',
+    'no_wrap': False,
+    'bos_text': '',
+    'key': 'content',
+}
+
+
+def is_iterable(obj: Any) -> bool:
+    """Check if obj is iterable."""
+    return issubclass(type(obj), Iterable)
+
+
+def do_merge_index(partitions: Iterable,
+                   mds_path: Union[str, Tuple[str, str]],
+                   keep_local: bool = True,
+                   skip: bool = False):
+    """Merge index.json from partitions into one for streaming."""
+    if not partitions or skip:
+        return
+
+    shards = []
+
+    for row in partitions:
+        mds_partition_index = f'{row.mds_path}/index.json'
```

Comment:
Can we have a global name for `index.json` and then use that everywhere? such as `MDS_META_FILE` or better name !!

Reply:
Yes~

Reply:
done.